azurerm/internal/services/loganalytics/log_analytics_cluster_resource.go

```diff
@@ -0,0 +1,300 @@
+			_, err := parse.LogAnalyticsClusterID(id)
+			return err
+		}),
+
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ForceNew:     true,
+				ValidateFunc: validate.LogAnalyticsClustersName,
+			},
+
+			"resource_group_name": azure.SchemaResourceGroupName(),
+
+			"location": azure.SchemaLocation(),
+
+			"identity": {
+				Type:     schema.TypeList,
+				Required: true,
+				ForceNew: true,
+				MaxItems: 1,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"type": {
+							Type:     schema.TypeString,
+							Required: true,
+							ForceNew: true,
+							ValidateFunc: validation.StringInSlice([]string{
+								string(operationalinsights.SystemAssigned),
+							}, false),
+						},
+
+						"principal_id": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+
+						"tenant_id": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+					},
+				},
+			},
+
+			// Per the documentation cluster capacity must start at 1000 GB and can go above 3000 GB with an exception by Microsoft
+			// so I am not limiting the upperbound here by design
+			// https://docs.microsoft.com/en-us/azure/azure-monitor/platform/manage-cost-storage#log-analytics-dedicated-clusters
+			"size_gb": {
+				Type:     schema.TypeInt,
+				Optional: true,
+				Default:  1000,
+				ValidateFunc: validation.All(
+					validation.IntAtLeast(1000),
+					validation.IntDivisibleBy(100),
+				),
+			},
+
+			"cluster_id": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+
+			"tags": tags.Schema(),
+		},
+	}
+}
+
+func resourceArmLogAnalyticsClusterCreate(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*clients.Client).LogAnalytics.ClusterClient
+	subscriptionId := meta.(*clients.Client).Account.SubscriptionId
+	ctx, cancel := timeouts.ForCreate(meta.(*clients.Client).StopContext, d)
+	defer cancel()
+
+	name := d.Get("name").(string)
+	resourceGroup := d.Get("resource_group_name").(string)
+
+	id := parse.NewLogAnalyticsClusterId(name, resourceGroup)
+
+	existing, err := client.Get(ctx, resourceGroup, name)
+	if err != nil {
+		if !utils.ResponseWasNotFound(existing.Response) {
+			return fmt.Errorf("checking for presence of existing Log Analytics Cluster %q (Resource Group %q): %+v", name, resourceGroup, err)
+		}
+	}
+	if existing.ID != nil && *existing.ID != "" {
+		return tf.ImportAsExistsError("azurerm_log_analytics_cluster", *existing.ID)
+	}
+
+	sku := &operationalinsights.ClusterSku{
+		Capacity: utils.Int64(int64(d.Get("size_gb").(int))),
+		Name:     operationalinsights.CapacityReservation,
+	}
+
+	parameters := operationalinsights.Cluster{
+		Location: utils.String(location.Normalize(d.Get("location").(string))),
+		Identity: expandArmLogAnalyticsClusterIdentity(d.Get("identity").([]interface{})),
+		Sku:      sku,
+		Tags:     tags.Expand(d.Get("tags").(map[string]interface{})),
+	}
```

Comment:
this could be
```suggestion
	parameters := operationalinsights.Cluster{
		Location: utils.String(location.Normalize(d.Get("location").(string))),
		Identity: expandArmLogAnalyticsClusterIdentity(d.Get("identity").([]interface{})),
		Sku:      sku := &operationalinsights.ClusterSku{
			Capacity: utils.Int64(int64(d.Get("size_gb").(int))),
			Name:     operationalinsights.CapacityReservation,
		},
		Tags:     tags.Expand(d.Get("tags").(map[string]interface{})),
	}
```